Mathlib/FieldTheory/SimpleExtension.lean

```diff
@@ -0,0 +1,120 @@
+/-
+Copyright (c) 2024 Daniel Weber. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Daniel Weber
+-/
+import Mathlib.FieldTheory.PrimitiveElement
+import Mathlib.RingTheory.Henselian
+import Mathlib.Algebra.Lie.OfAssociative
+
+/-!
+# Simple field extensions
+
+We define a typeclass `SimpleExtension` for one field being a simple extension of another
+(this is data because it contains which element was adjoined).
+
+We also define `TranscendentalExtension F K`, and show that it implies that `K` is `F[X]`'s fraction
+field.
+-/
+
+open Polynomial IntermediateField algebraMap
+
+/--
+A typeclass for K being a simple extension of F, F⟮k⟯ = K.
+-/
+class SimpleExtension (F K : Type*) [CommSemiring F] [Field K] [Algebra F K] where
+  /-- The element of K which is adjoined to F. -/
+  k : K
+  /-- Every value in K is a fraction of two polynomials with coefficients in F, evaluated at k. -/
+  is_adjoin' : ∀ (x : K), ∃ (r s : Polynomial F), x = (aeval k) r / (aeval k) s
```

Comment:
I think `class IsSimpleExtension ... : Prop` is better? Your current definition is a `Type` which carries a data `k` which is the generator of `K` over `F`. The disadvantage is that different `k` gives different instance `SimpleExtension F K`, which in my opinion complicates things.

Reply:
My use for this `SimpleExtension` is for Liouville's theorem, where I extend this with the existence of a `v` in `F` such that `k' / k = v'` (an exponential extension) or `k' = v' / v` (a logarithmic extension). Would there be a good way to do that with `IsSimpleExtension`?

Reply:
I'm not sure. But from your code `Algebra F[X] K` seems that the generator `k` needs to be gives, otherwise this instance cannot be defined. So maybe your `SimpleExtension` with data is better...

I think you should clearly state in the docstring that `SimpleExtension` is with a generator given.

Reply:
I edited the docstring, is it better now?